### ISSUE TITLE
Manage exceptions for when activities don't exist to handle intents.

### ISSUE THREAD

**Issue by JotraN** [iSoron/uhabits#181]
Not a big problem, but the application currently crashes whenever there isn't an activity to handle a given intent (e.g., no email client when trying to send feedback to the developer). Using `resolveAcitivity`, as shown in this [example](https://developer.android.com/guide/components/intents-filters.html#ExampleSend), would probably be helpful along with some error message.

**Comment by iSoron** [comment]
Yes, instead of crashing, the app should show an error message. For the backups, if there is no app able to deal with the file after we export it, then it is probably better to just say that the backup has been saved on the SD card.

**Comment by JotraN** [comment]
Sounds good. You mind if I start working on this?

**Comment by iSoron** [comment]
Please, go ahead. I have assigned this issue to you.

**Comment by JotraN** [comment]
Hi, I'm running into some small issues that I'm not quite sure what the best approach to handling them would be.

First, some of the intents started in the Settings fragment are done via the preferences xml file - I can't use resolveActivity on these intents unless I move them to SettingsFragment and handle interactions there (which is kind of awkward since other intents in SettingsFragment are handled using setResultOnPreferenceClick and ListsHabitsScreen).

Second, AboutRootView doesn't contain a snackbar or a screen to display messages, should I just copying the snackbar code from BaseScreen?

**Comment by iSoron** [comment]
> I can't use resolveActivity on these intents unless I move them to SettingsFragment and handle interactions there 

I am okay with that. Another solution would be to check, inside the method `SettingsFragment#onPreferenceTreeClick(Preference)`, if `preference.getIntent()` can be resolved.

> Second, AboutRootView doesn't contain a snackbar or a screen to display messages

We create a BaseScreen [here][1]. You can make the constructor of `AboutRootView` receive a reference to this screen. I might unify BaseScreen and BaseRootView later, since their distinction is not very clear.

[1]: https://github.com/iSoron/uhabits/blob/638b3f763cda02dd8d77d1ff0f456ba733d6790e/app/src/main/java/org/isoron/uhabits/activities/about/AboutActivity.java#L38

**Comment by JotraN** [comment]
Hello,

Thanks for the help. I'm just about done, and decided to handle the settings related intents in the `onPreferenceTreeClick` function, however I ran into the snackbar issue with the SettingsFragment as well.

I was able to workaround it by creating a `BaseScreen` in the fragment's `onCreate`. Alternatively I could also create it in the activity and pass it in like your suggestion for the `AboutRootView`. I'm not sure which approach is better since `SettingsFragment` extends `PreferenceFragmentCompat` instead of a `BaseRootView` or `BaseScreen`.
```
screen = new BaseScreen((BaseActivity) getActivity());
        screen.setRootView(new BaseRootView(getContext()) {
            @NonNull
            @Override
            public Toolbar getToolbar() {
                return toolbar;
            }
        });
```

**Comment by devurandom** [comment]
@jotran were you able to finish this?

**Comment by JotraN** [comment]
@devurandom Nope, I got distracted. Feel free to pick it up if you want.
